pkg/flink/internal/store/store.go

```diff
@@ -234,6 +256,27 @@ func (s *Store) waitForPendingStatement(ctx context.Context, statementName strin
 	}
 }
 
+func (s *Store) getStatusDetail(statementObj flinkgatewayv1beta1.SqlV1beta1Statement) string {
+	status := statementObj.GetStatus()
+	if status.GetDetail() != "" {
+		return status.GetDetail()
+	}
+
+	// if the status detail field is empty, we check if there's an exception instead
+	exceptionsResponse, err := s.authenticatedGatewayClient().GetExceptions(s.appOptions.GetEnvironmentId(), statementObj.GetName(), s.appOptions.GetOrgResourceId())
+	if err != nil {
+		return ""
+	}
+
+	exceptions := exceptionsResponse.GetData()
+	if len(exceptions) < 1 {
```

Comment:
This seems more readable since we can't have negative lengths
```suggestion
	if len(exceptions) == 0 {
```